sig-architecture/api_changes.md

```diff
@@ -0,0 +1,71 @@
+# Changing the API
+
+This document is oriented at developers who want to change existing APIs for Open Cluster Management. As Open Cluster Management uses [Kubernetes Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) to define its own API resources, a set of API change guidelines will also be applied to any [Kubernetes Custom Resource Definitions].
+
+## Contents
+
+- [Things you should consider before you change the APIs](#things-you-should-consider-before-you-change-the-apis)
+- [Versions in the APIs](#versions-in-the-apis)
+- [How to Add a New Version](#how-to-add-a-new-version)
+- [Migrate stored objects to the new version](#migrate-stored-objects-to-the-new-version)
+
+### Things you should consider before you change the APIs
+
+Before attempting a change to the API, you should familiarize yourself with a number of [existing API types](https://github.com/open-cluster-management/api) and with the API conventions.
+
+You should also consider the compatibility of the API changes. Open Cluster Management considers forwards and backwards compatibility of its APIs a top priority.
+
+An API change is considered compatible if it:
+
+- adds new functionality that is not required for correct behavior (e.g., does not add a new required field)
```

Comment:
I think we should add some instruction on what if we want to change the semantics.

Reply:
Added an example, PTAL https://github.com/open-cluster-management/community/pull/27/commits/f4d93cd63d39caf2c4272369fab469ba51b3c110